src/java/org/apache/cassandra/service/TraceLogSpanObserver.java

```diff
@@ -0,0 +1,82 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.ZonedDateTime;
+import java.util.stream.Collectors;
+import com.palantir.sls.logging.TraceLogs;
+import com.palantir.sls.spec.logging.Annotation;
+import com.palantir.sls.spec.logging.TraceLogV1;
+import org.apache.logging.log4j.message.ObjectMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.palantir.tracing.api.Span;
+import com.palantir.tracing.api.SpanObserver;
+import java.time.Clock;
+
+public class TraceLogSpanObserver implements SpanObserver
+{
+    private static final Logger logger = LoggerFactory.getLogger(TraceLogSpanObserver.class);
+    private final PrintWriter output;
+
+    public TraceLogSpanObserver() throws IOException
+    {
+        FileWriter fw = new FileWriter("cassandra-trace.log", true);
+        this.output = new PrintWriter(new BufferedWriter(fw));
+    }
+
+
+    public void consume(Span span)
+    {
+        try
+        {
+            TraceLogV1 v1 =  TraceLogs.v1("cassandra", span, () -> ZonedDateTime.now(Clock.systemUTC()));
+            logger.trace(new ObjectMessage(v1).toString());
+            output.println(convert(v1));
+            logger.info("finished writing span");
+            output.flush();
+        } catch (Exception e) {
+            logger.warn("Failed to write span to disk", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String convert(TraceLogV1 trace) {
+        com.palantir.sls.spec.logging.Span span = trace.getSpan();
+       String annotations = span.getAnnotations().stream()
+                                 .map(this::convertAnnotation).collect(Collectors.joining(","));
+
+       return String.format("{\"type\":\"trace.1\",\"time\":\"%s\",\"unsafeParams\":{}," +
+       "\"span\":{\"traceId\":\"%s\",\"id\":\"%s\",\"name\":\"%s\"," +
+       "\"parentId\":\"%s\",\"timestamp\":%s,\"duration\":%s,\"annotations\":[%s]," +
+       "\"tags\":{}}}", trace.getTime(), span.getTraceId(), span.getId(), span.getName(), span.getParentId().orElse(""),
+                            span.getTimestamp(), span.getDuration(), annotations);
```

Comment:
Do we need to ensure any of these are encoded appropriately for json?

`String.format` can be slow as each invocation has to parse the format string, so we might want to consider just string concatenation, see https://www.javaspecialists.eu/archive/Issue294-String.format-3x-faster-in-Java-17.html


```suggestion
       return "{\"type\":\"trace.1\",\"time\":\"" + trace.getTime() + "\",\"unsafeParams\":{}," +
       "\"span\":{\"traceId\":\"" + span.getTraceId() + "\",\"id\":\"" + span.getId() + "\",\"name\":\"" + span.getName() + "\"," +
       "\"parentId\":\"" + span.getParentId().orElse("") + "\",\"timestamp\": " + span.getTimestamp() + ",\"duration\": " + span.getDuration() + ",\"annotations\":[" + annotations + "]," +
       "\"tags\":{}}}";
```

Reply:
I got ObjectMapper working so this is no longer needed